proposals/0000-hybrid-output.md

```diff
@@ -0,0 +1,139 @@
+- Start Date: 2022-10-25
+- Reference Issues: TODO
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+Currently, Astro's [`output`](https://docs.astro.build/en/reference/configuration-reference/#output) enables developers to switch between `'static'` (default) and `'server'` outputs. One of our most requested features is to enable route-level control of `output` so that some routes can be server-rendered while others are statically generated.
+
+# Example
+
+This RFC proposes a new, opt-in `output` mode named `'hybrid'`.
+
+```js
+import { defineConfig } from 'astro/config'
+import node from '@astrojs/node'
+
+export default defineConfig({
+  output: 'hybrid',
+  adapter: node()
+})
+```
+
+For individual routes, `'static'` will be the default. To opt-in to `'server'` output, users may `export const output: 'static' | 'server'`.
+
+```astro
+---
+export const output = 'server'
```

Comment:
will it be possible for our language tools to give type checking and auto-complete as an `export const`?

Reply:
Probably possible! Can add a note about this.

Reply:
Confirming that it's possible! We'll have to either provide a type for it, or fake completions / diagnostics (like NextJS does)